### PR TITLE
vaccination changes and hic pdf"

### PR DESCRIPTION
**Pull request by AmalAnil97**
vaccination changes and hic pdf"

### CODE REVIEW

File: src/views/EmployeeRecord/VacccinationInformation/Entrydetails.jsx [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+                : 'N/A'}/>
+        </Box>
+        <Box
+          borderRight={0.5}
+          borderLeft={0.5}
+          borderBottom={0.5}
+          borderColor="#C9CCD5"
+          sx={{ p: 1, display: 'flex', fontWeight: 400, width: '25%', height: 'auto' }}
+        >
+          <Heading title= {details?.third_dose_due_date ? formatWithSunday(details?.third_dose_due_date) : 'N/A'}/>
```
Comment (itmedicity) [inline src/views/EmployeeRecord/VacccinationInformation/Entrydetails.jsx:261]: Change  the fun passing functionality

File: src/views/EmployeeRecord/VacccinationInformation/FirstdoseNottaken.jsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
+import { Box, } from '@mui/joy'
+import React, { useEffect, useState } from 'react'
+import CommonAgGrid from 'src/views/Component/CommonAgGrid'
+import { lazy } from 'react'
+import AddTaskIcon from '@mui/icons-material/AddTask'
+import { Tooltip, Paper, } from '@mui/material'
+import BeenhereIcon from '@mui/icons-material/Beenhere'
+import { memo } from 'react'
+import { IconButton as OpenIcon } from '@mui/material'
+import { useCallback } from 'react'
```
Comment (itmedicity) [inline src/views/EmployeeRecord/VacccinationInformation/FirstdoseNottaken.jsx:10]: remove the multiple import

**Review by itmedicity — CHANGES_REQUESTED** [review]

File: src/views/EmployeeRecord/EmployeeFile/EmployeeProfile/EmpMenus/VaccinationInformation/VaccinationInformation.jsx [hunk 3]
```diff
@@ -1,30 +1,465 @@
...
+  return (
+    <Fragment>
+      <Box sx={{ width: '100%' }}>
+        <Paper sx={{ height: 30, display: 'flex', alignItems: 'center', p: 1 }}>
+          <DragIndicatorOutlinedIcon color="success" />
+          <Typography sx={{ fontWeight: 600, pt: 0.5 }}>Vaccination information</Typography>
+        </Paper>
+
+        <Box sx={{ p: 1, fontWeight: 500 }}>
+          <CssVarsProvider>
```
[the remark this answers is not in the document]
Comment (AmalAnil97) [reply src/views/EmployeeRecord/EmployeeFile/EmployeeProfile/EmpMenus/VaccinationInformation/VaccinationInformation.jsx]: correct

File: src/views/EmployeeRecord/EmployeeFile/EmployeeProfile/EmpMenus/VaccinationInformation/VaccinationInformation.jsx [hunk 4]
```diff
@@ -1,30 +1,465 @@
...
+              fontWeight: 400,
+              width: '25%',
+              height: 'auto',
+              textAlign: 'center',
+              flexDirection: 'column',
+            }}
+          >
+            <Box sx={{ display: 'flex', flexDirection: 'row' }}>
+              <Box borderBottom={1} borderRight={1} sx={{ width: '50%', height: 23 }}>
+                <CssVarsProvider>
```
[the remark this answers is not in the document]
Comment (AmalAnil97) [reply src/views/EmployeeRecord/EmployeeFile/EmployeeProfile/EmpMenus/VaccinationInformation/VaccinationInformation.jsx]: new component added

File: src/views/EmployeeRecord/EmployeeFile/EmployeeProfile/EmpMenus/VaccinationInformation/VaccinationInformation.jsx [hunk 5]
```diff
@@ -1,30 +1,465 @@
...
-        </Fragment>
-    )
+  const [data, setdata] = useState([])
+
+  const state = useSelector(
+    (state) => state.getPrifileDateEachEmp.empPersonalData.personalData,
+    _.isEqual,
+  )
+
+  const [vaccinationList, setVaccinationList] = useSelector(
```
[the remark this answers is not in the document]
Comment (AmalAnil97) [reply src/views/EmployeeRecord/EmployeeFile/EmployeeProfile/EmpMenus/VaccinationInformation/VaccinationInformation.jsx]: changed

File: src/views/EmployeeRecord/EmployeeFile/EmployeeProfile/EmpMenus/VaccinationInformation/VaccinationInformation.jsx [hunk 6]
```diff
@@ -1,30 +1,465 @@
...
+                <CssVarsProvider>
+                  <Typography sx={{}}>Given Date</Typography>
+                </CssVarsProvider>
+              </Box>
+            </Box>
+            <Box sx={{ display: 'flex', flexDirection: 'row' }}>
+              <Box borderBottom={1} borderRight={1} sx={{ width: '50%', height: 43 }}>
+                <CssVarsProvider>
+                  <Typography sx={{ p: 1 }}>
+                    {vaccinationList.second_dose_due_date === null
```
[the remark this answers is not in the document]
Comment (AmalAnil97) [reply src/views/EmployeeRecord/EmployeeFile/EmployeeProfile/EmpMenus/VaccinationInformation/VaccinationInformation.jsx]: resolved

File: src/views/EmployeeRecord/EmployeeFile/EmployeeProfile/EmpMenus/VaccinationInformation/Previousvacc.jsx [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+                p: 1,
+                display: 'flex',
+                width: '25%',
+                height: 'auto',
+                alignItems: 'center',
+                flexDirection: 'column',
+              }}
+            >
+              {/* FIRST DOSE DATE  */}
+              {item?.firstdose_date === null ? 'not required' : item.firstdose_date}
```
[the remark this answers is not in the document]
Comment (AmalAnil97) [reply src/views/EmployeeRecord/EmployeeFile/EmployeeProfile/EmpMenus/VaccinationInformation/Previousvacc.jsx]: added ?

File: src/views/EmployeeRecord/EmployeeFile/EmployeeProfile/EmpMenus/VaccinationInformation/Previousvacc.jsx [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+                p: 1,
+                display: 'flex',
+                width: '25%',
+                height: 'auto',
+                alignItems: 'center',
+                flexDirection: 'column',
+              }}
+            >
+              {/* THIRD DOSE DATE */}
+              {item?.third_dose_given_date === null ? 'not required' : item.third_dose_given_date}
```
[the remark this answers is not in the document]
Comment (AmalAnil97) [reply src/views/EmployeeRecord/EmployeeFile/EmployeeProfile/EmpMenus/VaccinationInformation/Previousvacc.jsx]: added

File: src/views/EmployeeRecord/VacccinationInformation/Entrydetails.jsx (hunk 1 above)
Comment (AmalAnil97) [reply src/views/EmployeeRecord/VacccinationInformation/Entrydetails.jsx]: changed

File: src/views/EmployeeRecord/VacccinationInformation/FirstdoseNottaken.jsx (hunk 2 above)
Comment (AmalAnil97) [reply src/views/EmployeeRecord/VacccinationInformation/FirstdoseNottaken.jsx]: changed the multiple import

**Review by itmedicity — APPROVED** [review]

**Pull request merged**